code/src/LeafletProvider/DrawingTools/DrawMarker.ts

```diff
@@ -0,0 +1,144 @@
+/// <reference path="AbstractProviderTool.ts" />
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+namespace LeafletProvider.DrawingTools {
+    export class DrawMarker extends AbstractProviderTool<Configuration.DrawingTools.DrawMarkerConfig> {
+        private _defaultIcon: L.DivIcon;
+
+        constructor(
+            map: OSFramework.OSMap.IMap,
+            drawingTools: OSFramework.DrawingTools.IDrawingTools,
+            drawingToolsId: string,
+            type: string,
+            configs: Configuration.DrawingTools.DrawMarkerConfig
+        ) {
+            super(
+                map,
+                drawingTools,
+                drawingToolsId,
+                type,
+                new Configuration.DrawingTools.DrawMarkerConfig(configs)
+            );
+
+            this._defaultIcon = new L.DivIcon({
+                iconSize: [24, 40],
+                className: 'marker-leaflet-icon',
+                iconAnchor: [12, 40]
+            });
+        }
+
+        /**
+         * Sets the new icon on the Marker just by using the iconUrl
+         * The width and the height of the icon will use the one in the configs (if set) or will use the default image size
+         * The icon will be centered by x axis on the marker position but on the y axis it will appear right above it
+         */
+        private _createIcon(iconUrl: string): L.DivIcon {
+            let icon: L.DivIcon;
+            // If the iconUrl is not set or is empty, we should use the defaultIcon
+            if (iconUrl === '') {
```

Comment:
Do we need this if/else statement? Maybe set icon to defaultValue on line 36 and add one if if iconUrl Is not empty